docs/scos/dev/tutorials-and-howtos/advanced-tutorials/tutorial-howto-import-1m-of-products-and-make-it-searchable-and-shown-on-the-storefront.md

```diff
@@ -0,0 +1,70 @@
+* [Bulk approach](/docs/scos/dev/tutorials-and-howtos/advanced-tutorials/howto-import-1m-of-products-and-make-it-searchable-and-shown-on-the-storefront#bulk-approach)
+
+Each of these approaches has its own advantages and disadvantages, which we will try to discuss in the current document.
+
+### One-by-one approach
+<a href="#one-by-one-approach"></a>
+
+This approach is mainly used for importing small amounts of data, as for importing a single entity, DataImport must:
+
+
+<a href="#one-by-one-approach-steps"></a>
+* Check the imported data to ensure that it is complete and accurate, without errors or inconsistencies.
+* Prepare the data for importing a single entity, including related entities (for example, TaxSet IDs for products).
+* Import (save) the data to the database.
+
+These steps are performed for each individual entity in the dataset, which is the simplest and most convenient way of importing, as it is easy to implement and to track potential errors. In case of an unsuccessful import, it is easy to determine which dataset is causing the problems.
+
+However, as you may have noticed, this approach has may not fit performance requirements of the project, as it saves entities one by one, which leads to a large number of transactions and increasing the load on the database, which may become a bottleneck.
+
+### Bulk approach
+<a href="bulk-approach"></a>
+
+The bulk approach to data import in Spryker has the same [steps as One-by-one approach](/docs/scos/dev/tutorials-and-howtos/advanced-tutorials/howto-import-1m-of-products-and-make-it-searchable-and-shown-on-the-storefront#one-by-one-approach-steps) except for the last step: while DataImport will perform all necessary actions to prepare for import, it will not save the entities to the database until a certain amount of data sets have accumulated, and later it will use a raw SQL query to perform the actual data import operation.
+
+This approach way more performant, but it creates several difficulties, namely:
+
+* Tracking errors becomes more difficult because DataImport now writes multiple entities to the database simultaneously. If the data contains an error, it's hard to determine which data set caused the problem (this issue should be addressed during data validation).
+* You will need to maintain a complex raw SQL query and modify it in case of changes to the data structure or import functionality.
+* The SQL query structure will likely differ depending on the selected RDBMS (PostgreSQL/MySQL).
+
+### Raw SQL queries and Common Table Expressions (CTE)
+
+In the paragraph above, we have learned that a raw SQL query provides better performance when importing large volumes of data and is the preferred method for storing information in a database. However, there are several approaches to consider, and we will discuss two of them:
+
+* Common Table Expressions (CTE)
```

Comment:
CTEs are not as performant on MariaDB, as we have discussed. Maybe let's not mention that option?